### PR TITLE
Add `mx_benchmark.py`.

### PR DESCRIPTION
**Pull request by axel22**
Review by @gilles-duboscq

### CODE REVIEW

File: mx_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        Arguments:
+            benchmarks (list of strings): List of benchmark names.
+            vmargs (list of strings): List of VM arguments.
+            runargs (list of strings): List of benchmark arguments.
+
+        Returns:
+            object: A list of measurement results.
+
+            A measurement result is an object that can be converted into JSON and is
+            merged with root.
```
Comment (dougxc) [inline mx_benchmark.py]: What's `root`?

File: mx_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+                  scoring function (e.g. `"1.54"`)
+                - `metric.type` -- a string denoting the type of the metric
+                  (e.g. `"numeric"`)
+                - `metric.better` -- `"higher"` if higher is better, `"lower"` otherwise
+                - `metric.iteration` -- iteration number of the measurement (e.g. `0`)
+        """
+        raise NotImplementedError()
+
+
+def add_benchmark_suite(suite):
```
Comment (dougxc) [inline mx_benchmark.py]: Maybe we should adopt the naming convention of `bmsuite` for BenchmarkSuite typed variables to minimize the confusion with normal mx suites.

File: mx_benchmark.py (hunk 1 above)
Comment (axel22) [inline mx_benchmark.py]: Meant as the root data point object. I'll rewrite it to just `merged with the data point object`.

File: mx_benchmark.py (hunk 2 above)
Comment (axel22) [inline mx_benchmark.py]: Sounds good.

**Comment by axel22** [comment]
@gilles-duboscq PTAL

**Pull request merged**